### PR TITLE
Login-panel

### PR DESCRIPTION
**Pull request by Septias**
Also makes the design responsive

### CODE REVIEW

**Comment by Septias** [comment]
@dignifiedquire

File: src/app.rs [hunk 1]
```diff
@@ -56,15 +59,28 @@ impl App {
...
+            right_side: Some(WidgetOption::AddAccount(AddAccount::new())),
+            state,
         }
     }
 }
 
 impl epi::App for App {
     fn update(&mut self, ctx: &egui::Context, _frame: &mut epi::Frame) {
-        render_sidebar(ctx, self.state());
+        ctx.set_debug_on_hover(true);
```
Comment (dignifiedquire) [inline src/app.rs:73]: this shouldn't be enabled by default

File: src/state.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@ use crate::dc::types::{ChatList, Event, Log, MessageList, SharedState};
 //use crate::scheduler::Scheduler;
 
 pub struct AppState {
-    shared_state: Arc<RwLock<State>>,
+    pub shared_state: Arc<RwLock<State>>,
```
Comment (dignifiedquire) [inline src/state.rs:14]: this shouldn't be public, tehre is an accessor `.shared_state()` if you need it

**Comment by dignifiedquire** [comment]
can you post a screenshot of how this currently looks please?

**Comment by Septias** [comment]
Here is a screenshot:
image

**Comment by Septias** [comment]
Here is a screenshot of the current state:
Selection_002

**Comment by Septias** [comment]
TODO:
- [ ] Open panel with add-button on sidebar
- [ ] Add close-button

**Comment by dignifiedquire** [comment]
any reason not to make it a popup?

**Comment by Septias** [comment]
Not really :thinking: You'd rather have one ?

**Comment by dignifiedquire** [comment]
> Not really 🤔 You'd rather have one ?

I think so, seems to match existing patterns more

**Comment by Septias** [comment]
image
@dignifiedquire

File: src/app.rs [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+    widgets::{add_account::AddAccount, mainpanel::render_main_panel, sidebar::render_sidebar},
 };
 
+use egui::{FontData, FontDefinitions, FontFamily, Visuals};
+
+use std::fs;
+
 pub struct App {
     state: AppState,
+    add_account_panel: AddAccount,
```
Comment (dignifiedquire) [inline src/app.rs:12]: this doesn't need to be persisted here, does it?

Comment (dignifiedquire) [reply src/app.rs:12]: Then it needs to be an `Option` and set to `None` if the panel is not shown

Comment (Septias) [reply src/app.rs:12]: I tested the constructor with a log and it seems like it really just got called once

**Comment by Septias** [comment]
@dignifiedquire 
Weißt du, wie man zwei Buttons auf einer horizontalen Axe maximal außeinander plaziert? Also das einer am linken und der andere am rechten Rand ist.
Außerdem ist es atm möglich einen Account mehrerer male zu importierten. DC an sich blockiert das nicht und ich weiß nicht, wie man verhindern könnte. Fällt dir was ein?

Ansonsten ist es soweit mergable

**Comment by Septias** [comment]
Not going to get to back to this any time soon

**Pull request closed without merging**